e2e/test/scenarios/onboarding/home/homepage.cy.spec.js

```diff
@@ -141,6 +147,69 @@ describe("scenarios > home > homepage", () => {
   });
 });
 
+describe("scenarios > home > custom homepage", () => {
+  describe("setting custom homepage", () => {
+    beforeEach(() => {
+      restore();
+      cy.signInAsAdmin();
+    });
+
+    it("should give you the option to set a custom home page in settings", () => {
+      cy.visit("/admin/settings/general");
+      cy.findByTestId("custom-homepage-setting").findByRole("switch").click();
+
+      cy.findByTestId("custom-homepage-dashboard-setting")
+        .findByRole("button")
+        .click();
+
+      popover().findByText("Orders in a dashboard").click();
+
+      cy.get("main").findByText("Saved");
+
+      cy.findByRole("navigation").findByText("Exit admin").click();
+      cy.location("pathname").should("equal", "/dashboard/1");
+    });
+
+    it("should give you the option to set a custom home page using home page CTA", () => {
+      cy.visit("/");
+      cy.get("main").findByText("Customize").click();
+      modal()
+        .findByText(/Select a dashboard/i)
+        .click();
+      popover().findByText("Orders in a dashboard").click();
+      modal().findByText("Save").click();
+      cy.location("pathname").should("equal", "/dashboard/1");
+    });
+  });
+
+  describe("custom homepage set", () => {
+    beforeEach(() => {
+      restore();
+      cy.signInAsAdmin();
+      cy.request("PUT", "/api/setting/custom-homepage", { value: true });
+      cy.request("PUT", "/api/setting/custom-homepage-dashboard", { value: 1 });
+    });
+
+    it("should redirect you if you do not have permissions for set dashboard", () => {
+      cy.signIn("nocollection");
+      cy.visit("/");
+
+      cy.location("pathname").should("equal", "/");
+    });
+
+    it("should not redirect you if you do have permissions for set dashboard", () => {
```

Comment:
This spec checks the home page changes for admins which is not very different from some specs above. Ideally, we want to test larger user-flows in E2E specs, so I wonder if you think it is reasonable to add a few additional assertions to the tests above to replace this separate spec and the one above